addons/account_edi_extended/models/account_move.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields, _
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    edi_show_abandon_cancel_button = fields.Boolean(
+        compute='_compute_edi_show_abandon_cancel_button')
+    edi_error_message = fields.Html(compute='_compute_edi_error_count_message')
+    edi_error_level = fields.Selection(selection=[('info', 'Info'), ('warning', 'Warning'), ('error', 'Error')], compute='_compute_edi_error_count_message')
+
+    @api.depends(
+        'edi_document_ids',
+        'edi_document_ids.state',
+        'edi_document_ids.blocking_level',
+        'edi_document_ids.edi_format_id',
+        'edi_document_ids.edi_format_id.name')
+    def _compute_edi_web_services_to_process(self):
+        # OVERRIDE to take blocking_level into account
+        for move in self:
+            to_process = move.edi_document_ids.filtered(lambda d: d.state in ['to_send', 'to_cancel'] and d.blocking_level != 'error')
+            format_web_services = to_process.edi_format_id.filtered(lambda f: f._needs_web_services())
+            move.edi_web_services_to_process = ', '.join(f.name for f in format_web_services)
+
+    @api.depends(
+        'state',
+        'edi_document_ids.state',
+        'edi_document_ids.attachment_id')
+    def _compute_edi_show_abandon_cancel_button(self):
+        for move in self:
+            move.edi_show_abandon_cancel_button = any([doc.edi_format_id._needs_web_services()
+                                                       and doc.state == 'to_cancel'
+                                                       and move.is_invoice(include_receipts=True)
+                                                       and doc.edi_format_id._is_required_for_invoice(move)
+                                                       for doc in move.edi_document_ids])
```

Comment:
don't need to create another list, you can remove `[` and `]` since `any` works fine with generators